erpnext/projects/report/timesheet_billing_summary/timesheet_billing_summary.py

```diff
@@ -0,0 +1,223 @@
+    if filters.get('project'):
+        condition += f"and project = '{filters.get('project')}'"
+
+    timesheet_detail = frappe.db.sql(f"""Select parent , project , billing_hours  , from_time , to_time , hours , billing_rate, is_billable,
+                                     activity_type , task 
+                                  From `tabTimesheet Detail` 
+                                  Where {condition}""",as_dict = 1)
+
+ 
+    for row in timesheet_detail:
+        if timesheets_map.get(row.parent):
+            row.update(timesheets_map.get(row.parent))
+    
+    from itertools import groupby
+    
+    def key_func(k):
+        return k['name']
+   
+    INFO = sorted(timesheet_detail, key=key_func)
+    
+    
+    for key, value in groupby(INFO, key_func):
+        total_hours = 0
+        total_billing_hours = 0
+        total_amount = 0
+
+        for row in list(value):
+
+            from_time, to_time = filters.from_date, filters.to_date
+
+            if str(row.to_time) < from_time or str(row.from_time) > to_time:
+                continue
+
+            if str(row.from_time) > from_time:
+                from_time = row.from_time
+
+            if str(row.to_time) < to_time:
+                to_time = row.to_time
+
+            activity_duration, billing_duration = get_billable_and_total_duration(row, from_time, to_time)
+            
+            if not filters.get('show_timesheet_detail'):
+                total_hours += activity_duration
+                total_billing_hours += billing_duration
+                total_amount += billing_duration * flt(row.billing_rate)
+                
+            if filters.get('show_timesheet_detail'):
+                data.append({
+                    "project": row.get('project'),
+                    "employee": row.get('employee'),
+                    "employee_name": row.get('employee_name'),
+                    "timesheet": row.get('name'),
+                    "total_billable_hours": billing_duration,
+                    "total_hours": activity_duration,
+                    "amount": billing_duration * flt(row.billing_rate),
+                    "from_time":from_time,
+                    "to_time":to_time,
+                    "activity_type":row.activity_type,
+                    "task":row.task
+                })
+                
+        if total_hours:
+            data.append(
+                {
+                    "project": row.get('project'),
+                    "employee": row.get('employee'),
+                    "employee_name": row.get('employee_name'),
+                    "timesheet": row.get('name'),
+                    "total_billable_hours": total_billing_hours,
+                    "total_hours": total_hours,
+                    "amount": total_amount,
+                }
+            )
+
+    return data
+
+def get_billable_and_total_duration(activity, start_time, end_time):

```

Comment:
An identical function exists in the `billing_summary.py`, that can be used. Why copy paste?